### DISCUSSION TITLE
How to add / bundle an external dependency?

### DISCUSSION THREAD

**Discussion by theogravity** [alex8088/electron-vite#96] [Q&A] [answered]
I'm trying to use `@kjn/electron-typesafe-ipc` to simplify invoke / handle development.

I created a folder called `/src/ipc` where my IPC channels would be implemented.

In `preload/index.ts`, I'm calling `invokeIpcChannel` `@kjn/electron-typesafe-ipc`, but when I launch the app, I get the following error in the devtools:

<img width="704" alt="image" src="https://user-images.githubusercontent.com/855434/216703128-1a3de503-1da7-48dd-8788-2d168436135b.png">

My `electron.vite.config.ts` file looks like this 

```
import { defineConfig, externalizeDepsPlugin } from 'electron-vite';

export default defineConfig({
  main: {
    plugins: [externalizeDepsPlugin()],
  },
  preload: {
    plugins: [externalizeDepsPlugin()],
  },
});
```

(we don't use renderer since we load an external URL)

and I do have `"src/ipc/*"` as part of `include` in `tsconfig.node.json`.

**Comment by alex8088** [comment, marked as answer]
`@kjn/electron-typesafe-ipc` is a pure esm module.

``` js
 main: {
    plugins: [externalizeDepsPlugin({ exclude: [ '@kjn/electron-typesafe-ipc' ]}
  }
 ```

or

install to `devDependences`

see
https://evite.netlify.app/guide/dev.html#dependencies-vs-devdependencies
